beacon-chain/forkchoice/doubly-linked-tree/reorg_late_blocks.go

```diff
@@ -0,0 +1,80 @@
+package doublylinkedtree
+
+import (
+	"github.com/prysmaticlabs/prysm/v3/config/params"
+	"github.com/prysmaticlabs/prysm/v3/time/slots"
+)
+
+// ShouldOverrideFCU returns whether the current forkchoice head is weak
+// and thus may be reorged when proposing the next block.
+// This function should only be called if the following two conditions are
+// satisfied:
+// 1- It is immediately after receiving a block that may be subject to a reorg
+//
+//	or
+//	It is right after processAttestationsThreshold and we have processed the
+//	current slots attestations.
+//
+// 2- The caller has already called Forkchoice.Head() so that forkchoice has
+// been updated.
+// 3- The beacon node is serving a validator that will propose during the next
+// slot.
+//
+// This function only applies an heuristic to decide if the beacon will update
+// the engine's view of head with the parent block or the incoming block. It
+// does not guarantee an attempted reorg. This will only be decided later at
+// proposal time by calling GetProposerHead.
+func (f *ForkChoice) ShouldOverrideFCU() (override bool) {
```

Comment:
I think we are missing this:
```
    # Only re-org if we are proposing on-time.
    time_into_slot = (store.time - store.genesis_time) % SECONDS_PER_SLOT
    proposing_on_time = time_into_slot <= SECONDS_PER_SLOT // INTERVALS_PER_SLOT // 2
```

Reply:
Yeah, that can't really happen, we will only call this function after receiving a block and we are proposing during the next slot. The spec implementation allows you to reorg a late block that was supposed to be ready during N and that arrives during N+1. We will always reorg such blocks anyway.